<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="first_tb3"  default="tb3_0"/>
  <arg name="second_tb3" default="tb3_1"/>
  <arg name="third_tb3"  default="tb3_2"/>

  <arg name="second_tb3_topic" default="/tb3_1"/>
  <arg name="third_tb3_topic" default="/tb3_2"/>

  <arg name="first_tb3_x_pos" default="-12.0"/>
  <arg name="first_tb3_y_pos" default="-11.5"/>
  <arg name="first_tb3_z_pos" default=" 0.0"/>
  <arg name="first_tb3_yaw"   default=" 0.0"/>

  <arg name="second_tb3_x_pos" default="-11.5"/>
  <arg name="second_tb3_y_pos" default=" 1.5"/>
  <arg name="second_tb3_z_pos" default=" 0.0"/>
  <arg name="second_tb3_yaw"   default="-0.7"/>

  <arg name="third_tb3_x_pos" default=" 11.5"/>
  <arg name="third_tb3_y_pos" default="-1.5"/>
  <arg name="third_tb3_z_pos" default=" 0.0"/>
  <arg name="third_tb3_yaw"   default=" 2.0"/>

  <arg name="cmd_vel" default="/cmd_vel"/>
  <arg name="odom" default="/odom"/>
  <arg name="laser_scan" default="/scan"/>
  <arg name="blue_error" default="/blue_error"/>

  <param name="topic_vel" value="$(arg cmd_vel)"/>
  <param name="topic_odom" value="$(arg odom)"/>
  <param name="topic_laser_scan" value="$(arg laser_scan)"/>
  <param name="topic_error" value="$(arg blue_error)"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find handrecog)/worlds/building2.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="true"/>
    <arg name="debug" value="false"/>
  </include>  

  <group ns = "$(arg first_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg first_tb3) -x $(arg first_tb3_x_pos) -y $(arg first_tb3_y_pos) -z $(arg first_tb3_z_pos) -Y $(arg first_tb3_yaw) -param robot_description" />
  </group>

  <group ns = "$(arg second_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg second_tb3) -x $(arg second_tb3_x_pos) -y $(arg second_tb3_y_pos) -z $(arg second_tb3_z_pos) -Y $(arg second_tb3_yaw) -param robot_description" />
  </group>

  <group ns = "$(arg third_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg third_tb3) -x $(arg third_tb3_x_pos) -y $(arg third_tb3_y_pos) -z $(arg third_tb3_z_pos) -Y $(arg third_tb3_yaw) -param robot_description" />
  </group>

  <node name="Wonder_$(arg second_tb3)" pkg="handrecog" type="wonder_tb3.py" args="$(arg second_tb3_topic)" />
  <node name="Wonder_$(arg third_tb3)" pkg="handrecog" type="wonder_tb3.py" args="$(arg third_tb3_topic)" />
  <node name="FollowBlue_$(arg second_tb3)" pkg="handrecog" type="follow_blue.py" args="$(arg second_tb3_topic)" />
  <node name="FollowBlue_$(arg third_tb3)" pkg="handrecog" type="follow_blue.py" args="$(arg third_tb3_topic)" />

<!--
  <node name="Wonder_TB3_1" pkg="handrecog" type="wonder_tb3_1.py" />
  <node name="Wonder_TB3_2" pkg="handrecog" type="wonder_tb3_2.py" />
  <node name="FollowRed_TB3_1" pkg="handrecog" type="follow_blue.py" /> -->

</launch>
